app/Services/Contact/Avatar/UpdateAvatar.php

```diff
@@ -54,12 +54,15 @@ public function execute(array $data) : Contact
         }
 
         $contact->avatar_source = $data['source'];
-        $contact->avatar_photo_id = null;
-
-        // in case of a photo, set the photo as the avatar
-        if ($data['source'] === 'photo') {
-            $contact->avatar_photo_id = $data['photo_id'];
-            $contact->photos()->syncWithoutDetaching([$data['photo_id']]);
+        switch ($contact->avatar_source) {
+            case 'photo':
+            // in case of a photo, set the photo as the avatar
+                $contact->avatar_photo_id = $data['photo_id'];
+                $contact->photos()->syncWithoutDetaching([$data['photo_id']]);
+            break;
```

Comment:
I think we need a better indentation for the two `break` words.

Reply:
ok, fixed